diff -Naur shellinabox-2.10.orig/Makefile.am shellinabox-2.10/Makefile.am
--- shellinabox-2.10.orig/Makefile.am	2009-11-18 17:55:52.000000000 +0100
+++ shellinabox-2.10/Makefile.am	2010-03-25 14:16:03.000000000 +0100
@@ -107,6 +107,7 @@
 objcopyflags         = case "$(host_cpu)" in                                  \
                          i[0-9]86) echo '-O elf32-i386 -B i386';;             \
                          x86_64)   echo '-O elf64-x86-64 -B i386:x86-64';;    \
+                         mips*)    echo '-O elf32-tradbigmips -B mips:isa32';; \
                          arm*)     echo '-O elf32-littlearm -B arm';;         \
                        esac
 
diff -Naur shellinabox-2.10.orig/Makefile.in shellinabox-2.10/Makefile.in
--- shellinabox-2.10.orig/Makefile.in	2009-11-21 23:47:39.000000000 +0100
+++ shellinabox-2.10/Makefile.in	2010-03-25 14:16:28.000000000 +0100
@@ -371,6 +371,7 @@
 objcopyflags = case "$(host_cpu)" in                                  \
                          i[0-9]86) echo '-O elf32-i386 -B i386';;             \
                          x86_64)   echo '-O elf64-x86-64 -B i386:x86-64';;    \
+                         mips*)    echo '-O elf32-tradbigmips -B mips:isa32';; \
                          arm*)     echo '-O elf32-littlearm -B arm';;         \
                        esac
 
