--- a/deps/v8/src/base/logging.cc
+++ b/deps/v8/src/base/logging.cc
@@ -4,7 +4,7 @@
 
 #include "src/base/logging.h"
 
-#if V8_LIBC_GLIBC || V8_OS_BSD
+#if (V8_LIBC_GLIBC && !defined(__UCLIBC__)) || V8_OS_BSD
 # include <cxxabi.h>
 # include <execinfo.h>
 #elif V8_OS_QNX
@@ -51,7 +51,7 @@
 
 // Attempts to dump a backtrace (if supported).
 void DumpBacktrace() {
-#if V8_LIBC_GLIBC || V8_OS_BSD
+#if (V8_LIBC_GLIBC && !defined(__UCLIBC__)) || V8_OS_BSD
   void* trace[100];
   int size = backtrace(trace, arraysize(trace));
   char** symbols = backtrace_symbols(trace, size);
