--- a/deps/v8/src/sampler.cc
+++ b/deps/v8/src/sampler.cc
@@ -398,8 +398,8 @@ void SignalHandler::HandleProfilerSignal(int signal, siginfo_t* info,
   state.fp = reinterpret_cast<Address>(mcontext.regs[29]);
 #elif V8_HOST_ARCH_MIPS
   state.pc = reinterpret_cast<Address>(mcontext.pc);
-  state.sp = reinterpret_cast<Address>(mcontext.gregs[29]);
-  state.fp = reinterpret_cast<Address>(mcontext.gregs[30]);
+  state.sp = reinterpret_cast<Address>(mcontext.pgregs[29]);
+  state.fp = reinterpret_cast<Address>(mcontext.pgregs[30]);
 #elif V8_HOST_ARCH_MIPS64
   state.pc = reinterpret_cast<Address>(mcontext.pc);
   state.sp = reinterpret_cast<Address>(mcontext.gregs[29]);
