Index: oos-app-editor/components/install/process.php
===================================================================
--- oos-app-editor.orig/components/install/process.php
+++ oos-app-editor/components/install/process.php
@@ -154,7 +154,7 @@ define("BASE_URL", "' . $_SERVER["HTTP_H
 define("THEME", "default");
 
 // ABSOLUTE PATH
-define("WHITEPATHS", BASE_PATH . ",/home");
+define("WHITEPATHS", BASE_PATH . ",/home,/");
 
 // SESSIONS (e.g. 7200)
 $cookie_lifetime = "0";
