Index: oos-app-code-editor/lib/diff_match_patch.php
===================================================================
--- oos-app-code-editor.orig/lib/diff_match_patch.php
+++ oos-app-code-editor/lib/diff_match_patch.php
@@ -2071,6 +2071,8 @@ define('Match_MaxBits', PHP_INT_SIZE * 8
 function charCodeAt($str, $pos) {
 	return mb_ord(mb_substr($str, $pos, 1));
 }
+// fix for mb_ord error as per https://github.com/Codiad/Codiad/issues/1082
+/*
 function mb_ord($v) {
 	$k = mb_convert_encoding($v, 'UCS-2LE', 'UTF-8'); 
 	$k1 = ord(substr($k, 0, 1)); 
@@ -2080,6 +2082,7 @@ function mb_ord($v) {
 function mb_chr($num){
 	return mb_convert_encoding('&#'.intval($num).';', 'UTF-8', 'HTML-ENTITIES');
 }
+*/
 
 /**
  * as in javascript encodeURI() following the MDN description
