diff -rup fbida-2.09.orig/fbi.c fbida-2.09/fbi.c
--- fbida-2.09.orig/fbi.c	2012-02-22 18:13:36.000000000 +0100
+++ fbida-2.09/fbi.c	2015-01-20 11:14:33.768924953 +0100
@@ -145,6 +145,7 @@ int read_ahead;
 int editable;
 int blend_msecs;
 int perfmon = 0;
+int notty = 0;
 
 /* font handling */
 static char *fontname = NULL;
@@ -1464,6 +1465,8 @@ main(int argc, char *argv[])
     fontname    = cfg_get_str(O_FONT);
     filelist    = cfg_get_str(O_FILE_LIST);
     
+    notty		= GET_NOTTY();
+    
     if (filelist)
 	flist_add_list(filelist);
     for (i = optind; i < argc; i++)
diff -rup fbida-2.09.orig/fbiconfig.c fbida-2.09/fbiconfig.c
--- fbida-2.09.orig/fbiconfig.c	2012-02-22 18:13:36.000000000 +0100
+++ fbida-2.09/fbiconfig.c	2015-01-20 12:45:06.201142919 +0100
@@ -171,6 +171,11 @@ struct cfg_cmdline fbi_cfg[] = {
 	.desc     = "use video mode <arg> (from /etc/fb.modes)",
 
     },{
+	.cmdline  = "notty",
+	.option   = { O_NOTTY },
+	.value    = "1",
+	.desc     = "don't capture control tty (for usage over ssh, serial etc.)",
+    },{
 	/* end of list */
     }
 };
diff -rup fbida-2.09.orig/fbiconfig.h fbida-2.09/fbiconfig.h
--- fbida-2.09.orig/fbiconfig.h	2012-02-22 18:13:36.000000000 +0100
+++ fbida-2.09/fbiconfig.h	2015-01-20 11:49:14.733008448 +0100
@@ -36,6 +36,8 @@
 #define O_FONT                  O_OPTIONS, "font"
 #define O_VIDEO_MODE            O_OPTIONS, "video-mode"
 
+#define O_NOTTY            O_OPTIONS, "notty"
+
 #define GET_HELP()		cfg_get_bool(O_HELP,          0)
 #define GET_VERSION()		cfg_get_bool(O_VERSION,       0)
 #define GET_WRITECONF()		cfg_get_bool(O_WRITECONF,     0)
@@ -64,6 +66,8 @@
 
 #define GET_GAMMA()             cfg_get_float(O_GAMMA,        1)
 
+#define GET_NOTTY()		cfg_get_int(O_NOTTY,      0)
+
 /* -------------------------------------------------------------------------- */
 
 extern struct cfg_cmdline fbi_cmd[];
diff -rup fbida-2.09.orig/fbtools.c fbida-2.09/fbtools.c
--- fbida-2.09.orig/fbtools.c	2012-02-22 18:13:36.000000000 +0100
+++ fbida-2.09/fbtools.c	2015-01-20 11:25:22.780950993 +0100
@@ -88,6 +88,8 @@ static void dev_init(void)
 
 extern int debug;
 
+extern int notty;
+
 static void
 fb_switch_signal(int signal)
 {
@@ -108,6 +110,9 @@ fb_switch_signal(int signal)
 void
 fb_switch_release()
 {
+    if(notty)
+    	return;
+    	
     ioctl(tty, VT_RELDISP, 1);
     fb_switch_state = FB_INACTIVE;
     if (debug)
@@ -117,9 +122,12 @@ fb_switch_release()
 void
 fb_switch_acquire()
 {
-    ioctl(tty, VT_RELDISP, VT_ACKACQ);
-    fb_switch_state = FB_ACTIVE;
-    if (debug)
+    if(notty)
+    	return;
+    	
+	ioctl(tty, VT_RELDISP, VT_ACKACQ);
+	fb_switch_state = FB_ACTIVE;
+	if (debug)
 	write(2,"vt: acquire\n",12);
 }
 
@@ -133,19 +141,20 @@ fb_switch_init()
     sigemptyset(&act.sa_mask);
     sigaction(SIGUSR1,&act,&old);
     sigaction(SIGUSR2,&act,&old);
-    
-    if (-1 == ioctl(tty,VT_GETMODE, &vt_mode)) {
-	perror("ioctl VT_GETMODE");
-	exit(1);
-    }
-    vt_mode.mode   = VT_PROCESS;
-    vt_mode.waitv  = 0;
-    vt_mode.relsig = SIGUSR1;
-    vt_mode.acqsig = SIGUSR2;
-    
-    if (-1 == ioctl(tty,VT_SETMODE, &vt_mode)) {
-	perror("ioctl VT_SETMODE");
-	exit(1);
+    if(!notty) {
+		if (-1 == ioctl(tty,VT_GETMODE, &vt_mode)) {
+		perror("ioctl VT_GETMODE");
+		exit(1);
+		}
+		vt_mode.mode   = VT_PROCESS;
+		vt_mode.waitv  = 0;
+		vt_mode.relsig = SIGUSR1;
+		vt_mode.acqsig = SIGUSR2;
+		
+		if (-1 == ioctl(tty,VT_SETMODE, &vt_mode)) {
+		perror("ioctl VT_SETMODE");
+		exit(1);
+		}
     }
     return 0;
 }
@@ -249,6 +258,9 @@ fb_setvt(int vtno)
     struct vt_stat vts;
     char vtname[12];
     
+    if(notty)
+    	return;
+    
     if (vtno < 0) {
 	if (-1 == ioctl(tty,VT_OPENQRY, &vtno) || vtno == -1) {
 	    perror("ioctl VT_OPENQRY");
@@ -301,6 +313,9 @@ static int fb_activate_current(int tty)
 {
     struct vt_stat vts;
     
+    if(notty)
+    	return 0;
+    
     if (-1 == ioctl(tty,VT_GETSTATE, &vts)) {
 	perror("ioctl VT_GETSTATE");
 	return -1;
@@ -325,13 +340,16 @@ fb_init(char *device, char *mode, int vt
 
     dev_init();
     tty = 0;
-    if (vt != 0)
-	fb_setvt(vt);
-
-    if (-1 == ioctl(tty,VT_GETSTATE, &vts)) {
-	fprintf(stderr,"ioctl VT_GETSTATE: %s (not a linux console?)\n",
-		strerror(errno));
-	exit(1);
+    
+    if(!notty) {
+		if (vt != 0)
+		fb_setvt(vt);
+
+		if (-1 == ioctl(tty,VT_GETSTATE, &vts)) {
+		fprintf(stderr,"ioctl VT_GETSTATE: %s (not a linux console?)\n",
+			strerror(errno));
+		exit(1);
+		}
     }
     
     if (NULL == device) {
@@ -376,15 +394,18 @@ fb_init(char *device, char *mode, int vt
 	    exit(1);
 	}
     }
-    if (-1 == ioctl(tty,KDGETMODE, &kd_mode)) {
-	perror("ioctl KDGETMODE");
-	exit(1);
-    }
-    if (-1 == ioctl(tty,VT_GETMODE, &vt_omode)) {
-	perror("ioctl VT_GETMODE");
-	exit(1);
+    
+    if(!notty) {
+		if (-1 == ioctl(tty,KDGETMODE, &kd_mode)) {
+		perror("ioctl KDGETMODE");
+		exit(1);
+		}
+		if (-1 == ioctl(tty,VT_GETMODE, &vt_omode)) {
+		perror("ioctl VT_GETMODE");
+		exit(1);
+		}
+		tcgetattr(tty, &term);
     }
-    tcgetattr(tty, &term);
     
     /* switch mode */
     fb_setmode(mode);
@@ -441,11 +462,14 @@ fb_init(char *device, char *mode, int vt
 	    goto err;
 	}
     }
-    if (-1 == ioctl(tty,KDSETMODE, KD_GRAPHICS)) {
-	perror("ioctl KDSETMODE");
-	goto err;
+    
+    if(!notty) {
+		if (-1 == ioctl(tty,KDSETMODE, KD_GRAPHICS)) {
+		perror("ioctl KDSETMODE");
+		goto err;
+		}
+		fb_activate_current(tty);
     }
-    fb_activate_current(tty);
 
     /* cls */
     fb_memset(fb_mem+fb_mem_offset, 0, fb_fix.line_length * fb_var.yres);
@@ -460,8 +484,10 @@ void
 fb_cleanup(void)
 {
     /* restore console */
-    if (-1 == ioctl(tty,KDSETMODE, kd_mode))
-	perror("ioctl KDSETMODE");
+    if(!notty) {
+		if (-1 == ioctl(tty,KDSETMODE, kd_mode))
+		perror("ioctl KDSETMODE");
+	}
     if (-1 == ioctl(fb,FBIOPUT_VSCREENINFO,&fb_ovar))
 	perror("ioctl FBIOPUT_VSCREENINFO");
     if (-1 == ioctl(fb,FBIOGET_FSCREENINFO,&fb_fix))
@@ -473,14 +499,16 @@ fb_cleanup(void)
     }
     close(fb);
 
-    if (-1 == ioctl(tty,VT_SETMODE, &vt_omode))
-	perror("ioctl VT_SETMODE");
-    if (orig_vt_no && -1 == ioctl(tty, VT_ACTIVATE, orig_vt_no))
-	perror("ioctl VT_ACTIVATE");
-    if (orig_vt_no && -1 == ioctl(tty, VT_WAITACTIVE, orig_vt_no))
-	perror("ioctl VT_WAITACTIVE");
-    tcsetattr(tty, TCSANOW, &term);
-    close(tty);
+    if(!notty) {
+		if (-1 == ioctl(tty,VT_SETMODE, &vt_omode))
+		perror("ioctl VT_SETMODE");
+		if (orig_vt_no && -1 == ioctl(tty, VT_ACTIVATE, orig_vt_no))
+		perror("ioctl VT_ACTIVATE");
+		if (orig_vt_no && -1 == ioctl(tty, VT_WAITACTIVE, orig_vt_no))
+		perror("ioctl VT_WAITACTIVE");
+		tcsetattr(tty, TCSANOW, &term);
+		close(tty);
+    }
 }
 
 /* -------------------------------------------------------------------- */
diff -rup fbida-2.09.orig/GNUmakefile fbida-2.09/GNUmakefile
--- fbida-2.09.orig/GNUmakefile	2012-02-22 18:13:36.000000000 +0100
+++ fbida-2.09/GNUmakefile	2015-01-20 11:50:10.633010690 +0100
@@ -52,7 +52,8 @@ HAVE_LIBSANE	:= $(call ac_lib,sane_init,
 HAVE_LIBCURL	:= $(call ac_lib,curl_easy_init,curl)
 HAVE_LIBLIRC	:= $(call ac_lib,lirc_init,lirc_client)
 HAVE_MOTIF	:= $(call ac_lib,XmStringGenerate,Xm,-L/usr/X11R6/$(LIB) -lXpm -lXt -lXext -lX11)
-JPEG_VER        := $(call ac_jpeg_ver)
+#JPEG_VER        := $(call ac_jpeg_ver)
+JPEG_VER		:= 62
 endef
 
 # transposing
@@ -149,7 +150,7 @@ ida : LDFLAGS	+= -L/usr/X11R6/$(LIB)
 ida : LDLIBS	+= -lXm -lXpm -lXt -lXext -lX11
 
 # jpeg/exif libs
-ida : LDLIBS	+= -ljpeg -lexif -lm
+ida : LDLIBS	+= -ljpeg -lexif -lm -liconv
 
 # RegEdit.c is good old K&R ...
 RegEdit.o : CFLAGS += -Wno-missing-prototypes -Wno-strict-prototypes
@@ -181,11 +182,10 @@ OBJS_FBI += $(filter-out wr/%,$(call ac_
 # jpeg/exif libs
 fbi : CFLAGS += $(shell pkg-config --cflags freetype2 fontconfig)
 fbi : LDLIBS += $(shell pkg-config --libs   freetype2 fontconfig)
-fbi : LDLIBS += -ljpeg -lexif -lm
+fbi : LDLIBS += -ljpeg -lexif -lm -liconv
 
 fbi: $(OBJS_FBI) $(OBJS_READER)
 
-
 ########################################################################
 # general rules
 
